kubernetes-extension/src/main/java/org/ballerinax/kubernetes/processors/ServiceAnnotationProcessor.java

```diff
@@ -110,7 +110,20 @@ public void processAnnotation(SimpleVariableNode variableNode, AnnotationAttachm
 
     private int extractPort(BLangTypeInit bListener) throws KubernetesPluginException {
         try {
-            return Integer.parseInt(bListener.argsExpr.get(0).toString());
+            if ("ballerina/http:Listener".equals(bListener.type.toString())) {
```

Comment:
Lets change to check first position for the port regardless of the type of the listener then fallback to `else`.